src/otx/algorithms/common/configs/training_base.py

```diff
@@ -227,6 +216,16 @@ class BasePostprocessing(ParameterGroup):
             affects_outcome_of=ModelLifecycle.INFERENCE,
         )
 
+        max_num_detections = configurable_integer(
+            header="Maximum number of detection per image",
+            description="Extra detection outputs will be discared in non-maximum suppression process. "
+            "Defaults to 0, which means per-model default value.",
+            default_value=0,
+            min_value=0,
+            max_value=10000,
+            affects_outcome_of=ModelLifecycle.INFERENCE,
+        )
+
```

Comment:
Don't we have a parameter to control maximum outputs of object detector?

Reply:
I didn't know that, except for the `--tiling_parameters.tile_max_number`.
Could you let me know what it is?

Reply:
I think that's the parameter I ask. What is the relation between tile_max_number and max_num_detections?

Reply:
It's only for tile-merging NMS, which means it will restrict the merged output number of all tiles.
`max_num_detections` in this case controls per-tile detections. So in fact, it had been limited to detect maximum 200 objects for each tile so far.

Anyway, I think I need to align this 2 somehow... let me think.
@eugene123tw Do you have any suggestion?

Reply:
`--tiling_parameters.tile_max_number` is tiling-specific and there is an own adaptation mechanism so that I think we don't need to merge it with `--postprocessing.max_num_detections` for now because the latter would still control per-time maximum even in case of tiling.
I'd like to hear from @eugene123tw before resolving this conversation.

Reply:
One more question. If max_num_detections is 500 and a image has 4 tiles. What is the max number of outputs for this image if tile_max_number is 500. I think it should be 2000 but it would be 500

Reply:
Yes, that's the behavior. Currently, users need to set both values properly in case of "manual" tiling setting. In auto tiling, we set the value using dataset statistics.

Do you think the tile_max_number should be automatically adjusted to # tiles x max_num_detections? I'm a little bit against it.

Reply:
I don't think we should do it. It will raise complexity very much. Thanks